tests/unit/package_managers/test_pip.py

```diff
@@ -105,23 +112,25 @@ def test_get_pip_metadata(
     expect_name = toml_name or py_name or cfg_name
     expect_version = toml_version or py_version or cfg_version
 
-    if expect_name and expect_version:
+    if expect_name:
         name, version = pip._get_pip_metadata(PKG_DIR)
 
         assert name == expect_name
         assert version == expect_version
     else:
-        with pytest.raises(PackageRejected) as exc_info:
-            pip._get_pip_metadata(PKG_DIR)
-
-        if expect_name:
-            missing = "version"
-        elif expect_version:
-            missing = "name"
+        if repo_exists:
+            mock_get_repo_id.return_value = MOCK_REPO_ID
+            name, _ = pip._get_pip_metadata(PKG_DIR_SUBPATH)
+            assert name == "bar-subpath"
+            assert "Processing metadata from git repository" in caplog.text
```

Comment:
I noticed that we're not testing `version` at all in the else branch

Could we maybe refactor the test so that if `expect_name` is None, the test always expects an exception?

```python
expect_name = toml or py or cfg or origin
if expect_name:
    ...
else:
    with pytest.raises(...)
```